File: libraryCoreJava/src/main/java/com/juqitech/moretickets/core/http/HttpClient.java
package com.juqitech.moretickets.core.http;

import androidx.annotation.NonNull;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;

/**
 * @author zhiqiang
 * @date 2019-06-04
 * @desc
 */
public class HttpClient {

    public static <T> T create(@NonNull Class<T> clazz) {
        return getRetrofit().create(clazz);
    }

    public static Retrofit getRetrofit() {
        return RetrofitFactory.getInstance().getRetrofit();
    }

    public static OkHttpClient getOkHttpClient() {
        return RetrofitFactory.getInstance().getOkHttpClient();
    }

}

File: libraryCore/src/main/java/com/juqitech/moretickets/core/http/HttpClient.kt
package com.juqitech.moretickets.core.http

import okhttp3.OkHttpClient
import retrofit2.Retrofit

/**
 * @author zhanfeng
 * @date 2019-06-04
 * @desc
 */
object HttpClient {

    val retrofit: Retrofit
        get() = RetrofitFactory.instance.retrofit

    val okHttpClient: OkHttpClient
        get() = RetrofitFactory.instance.okHttpClient

    fun <T> create(clazz: Class<T>): T {
        return retrofit.create(clazz)
    }

}
